crates/detect-targets/src/detect/macos.rs

```diff
@@ -36,14 +36,20 @@ async fn is_arch_supported(arch_name: &str) -> bool {
 pub(super) async fn detect_alternative_targets(target: &str) -> impl Iterator<Item = String> {
     match target {
         AARCH64 => {
-            let is_x86_64_supported = is_arch_supported("x86_64").await;
+            // Spawn `arch` in parallel (probably from different threads if
+            // mutlti-thread runtime is used).
+            //
+            // These two tasks are never cancelled, so it can only fail due to
+            // panic, in which cause we would propagate by also panic here.
+            let x86_64h_task = tokio::spawn(is_arch_supported("x86_64h"));
+            let x86_64_task = tokio::spawn(is_arch_supported("x86_64"));
             [
                 // Prefer universal as it provides native arm executable
```

Comment:
Hmm unrelated to this PR but I wonder about that, given that universal can contain anything really. Like you could have a x64 and x64h universal with no arm

Reply:
Yeah, according to [bjorn3 on zulip](https://rust-lang.zulipchat.com/#narrow/stream/122651-general/topic/bad.20CPU.20type.20when.20executing.20.60x86_64h-apple-darwin.60.20binary/near/383618400) it could even have a PowerPC and m68k binary inside.